projects/js-packages/publicize-components/src/utils/share-status.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Normalizes the share status object.
+ *
+ * @param {import( './types' ).PostShareStatus} shareStatus - Share status object.
+ * @return {import( './types' ).PostShareStatus | undefined} - Normalized share status object.
+ */
+export function normalizeShareStatus( shareStatus ) {
+	if ( ! shareStatus || ! ( 'shares' in shareStatus ) || ! shareStatus.done ) {
+		return;
+	}
+	// Sort shares to show the latest shares on the top.
+	shareStatus.shares.sort( ( a, b ) => b.timestamp - a.timestamp );
```

Comment:
We need to return the value in every case.

```suggestion
	if ( shareStatus && 'shares' in shareStatus && ! shareStatus.done ) {
		// Sort shares to show the latest shares on the top.
		shareStatus.shares.sort( ( a, b ) => b.timestamp - a.timestamp );
	}
```